lesson_15/tdd/tdd_app/src/test/java/com/codedifferently/lesson15/EmployeeManagerTest.java

```diff
@@ -0,0 +1,89 @@
+package com.codedifferently.lesson15;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class EmployeeManagerTest {
+  private EmployeeManager employeeManager;
+
+  @BeforeEach
+  public void setUp() {
+    employeeManager = new EmployeeManager();
+  }
+
+  @Test
+  void testAddEmployee() {
+    // Arrange
+    Employee employee = new Employee(1, "John Doe", null, 0);
+
+    // Act
+    employeeManager.addEmployee(employee);
+
+    // Assert
+    assertEquals(1, employeeManager.getEmployeeCount());
+  }
+
+  @Test
+  void testUpdateEmployee() {
+    // Arrange
+    Employee employee = new Employee(1, "John Doe", null, 0);
+    Employee updatedEmployee = new Employee(1, "Jane Doe", null, 0);
+
+    // Act
+    employeeManager.addEmployee(employee);
+    employeeManager.updateEmployee(updatedEmployee);
+
+    // Assert
+    Employee retrievedEmployee = employeeManager.getEmployee(1);
+    assertEquals(updatedEmployee.getName(), retrievedEmployee.getName());
+  }
+
+  @Test
+  void testGetEmployee() {
+    // Arrange
+    Employee employee = new Employee(1, "John Doe", null, 0);
+
+    // Act
+    employeeManager.addEmployee(employee);
+    Employee retrievedEmployee = employeeManager.getEmployee(1);
+
+    // Assert
+    assertEquals(employee.getId(), retrievedEmployee.getId());
+    assertEquals(employee.getName(), retrievedEmployee.getName());
+  }
+
+  @Test
+  void testRemoveEmployee() {
+
+    Employee employee = new Employee(1, "John Doe", null, 0);
+
+    employeeManager.addEmployee(employee);
+    assertEquals(1, employeeManager.getEmployeeCount());
+
+    employeeManager.removeEmployee(1);
+    assertEquals(0, employeeManager.getEmployeeCount());
+  }
+
+  @Test
+  void testGetEmployeeCount() {
+
+    assertEquals(0, employeeManager.getEmployeeCount());
+
+    employeeManager.addEmployee(new Employee(1, "John Doe", null, 0));
+    employeeManager.addEmployee(new Employee(2, "Jane Doe", null, 0));
+
+    assertEquals(2, employeeManager.getEmployeeCount());
+  }
+
+  @Test
+  void testGetEmployeeThrowsExceptionForNonExistentId() {
```

Comment:
Help from chatGPT to understand how you check if an assertion was thrown. In fact this is familiar from one of the lectures when there was an example -- mostly from an amusing comment when anthony was working on a different test!